### PR TITLE
Fix K80 gpunode by correcting GCP image version

### PR DESCRIPTION
**Pull request by infwinston**
Bug report from Daniel: `nvidia-smi` doesn't work on `sky gpunode --cloud gcp`.
```
NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.
```

Reason: K80 requires nvidia-driver version to be <=470. (see details [here](https://cloud.google.com/compute/docs/gpus/install-drivers-gpu))
Before it worked fine as the cuda version was older on GCP images but several days ago GCP upgrades the driver version to 510 for all the deep learning images (see release note [here](https://cloud.google.com/deep-learning-vm/docs/release-notes)). From the note
```
The latest NVIDIA driver version does not support K80 GPUs. To use K80 GPUs, you must use an M94 or earlier environment.
```

This PR fixes this issue by using a previous version of `common-cu113` for K80.
For others, I'm not sure whether we should continue using the latest version of `common-cu113` which is subject to change or choose a static one. May need more discussion.

Note: to query older versions of GCP image try:
```
gcloud compute images list --project deeplearning-platform-release --show-deprecated
```

### CODE REVIEW

**Review by WoosukKwon — APPROVED** [review]
Thanks for identifying the cause of the error. LGTM.

nit: While hardcoding the image version works as a hot fix to the Daniel's problem, I'm a bit worried that it will add maintenance burden to us. Can we automate the image versioning?

**Comment by infwinston** [comment]
@WoosukKwon Could you say a bit more what version automation you referred to?

**Comment by WoosukKwon** [comment]
@infwinston It was more like a random thought about how to save our labor, because I feel this is a repeating issue. I don't have a concrete idea on this at the moment though.

**Comment by infwinston** [comment]
I see. I think what this PR does is to make GCP's behavior aligned with AWS/Azure where static images are used. Now we always use the latest version of `common-cu113` so things can break. With static version it should be more robust. 
However, I can imagine if there's a new accelerator requiring newer version of images then we have to manually specify one for it. But I don't see an easy solution here either..

File: sky/clouds/gcp.py [hunk 1]
```diff
@@ -216,8 +215,12 @@ def make_deploy_resources_variables(
                 # https://cloud.google.com/compute/docs/gpus
                 resources_vars['gpu'] = 'nvidia-tesla-{}'.format(acc.lower())
                 resources_vars['gpu_count'] = acc_count
-                # CUDA driver version 470.103.01, CUDA Library 11.3
-                image_id = _IMAGE_ID_PREFIX + 'common-cu113'
+                if acc == 'K80':
```
Comment (concretevitamin) [inline sky/clouds/gcp.py:218]: acc.upper()? Or would current PR work for `sky gpunode --gpus k80`?

Comment (infwinston) [reply sky/clouds/gcp.py:218]: Thanks, great catch. in fact this also catches another error when searching acc in region.
```
(sky) weichiang@mbp sky % sky gpunode --gpus k80 --cloud gcp
ValueError: Accelerator "k80" is not available in "us-central1" region/zone.
``` 

But is this still the case after we ship https://github.com/skypilot-org/skypilot/pull/1075? @WoosukKwon

Comment (infwinston) [reply sky/clouds/gcp.py:218]: Just tested this PR with latest master, it works well with
```
if acc == 'K80':
```

Comment (WoosukKwon) [reply sky/clouds/gcp.py:218]: Yeah after #1075 the internal name should be always 'K80' regardless of the user input.

**Comment by infwinston** [comment]
@WoosukKwon @concretevitamin let me know if you guys have more concerns otherwise I think we can merge.

**Comment by WoosukKwon** [comment]
@infwinston I don't have any particular concern on this PR. Let's ship it.

**Pull request merged**